### PR TITLE
PRM-2746 Retry notifications we have not received a FileReceived callback from SDES

### PR DESCRIPTION
**Pull request by AndrewRButcher**

### CODE REVIEW

File: app/scheduler/SchedulingActor.scala [hunk 1]
```diff
@@ -40,5 +40,5 @@ object SchedulingActor {
   case class MonitoringJobMessage(service: MonitoringJobService) extends ScheduledMessage[Either[ScheduleStatus.JobFailed, Seq[String]]]
   case class SendFileNotificationsToSDESMessage(service: SendFileNotificationsToSDESService)
     extends ScheduledMessage[Either[ScheduleStatus.JobFailed, String]]
-  case class HandleNotProcessedFilesFromSDESMessage(service: HandleNotProcessedFilesService) extends ScheduledMessage[Either[ScheduleStatus.JobFailed, String]]
+  case class HandleNotProcessedFilesFromSDESMessage(service: HandleStuckNotificationsService) extends ScheduledMessage[Either[ScheduleStatus.JobFailed, String]]
```
Comment (LewisMitchell) [inline app/scheduler/SchedulingActor.scala:43]: Could the message change as well?
```suggestion
  case class HandleStuckNotificationsMessage(service: HandleStuckNotificationsService) extends ScheduledMessage[Either[ScheduleStatus.JobFailed, String]]
```

File: app/scheduler/jobs/HandleNotProcessedFilesFromSDESJob.scala [hunk 2]
```diff
@@ -22,13 +22,13 @@ import play.api.Configuration
 import play.api.inject.ApplicationLifecycle
 import scheduler.ScheduledJob
 import scheduler.SchedulingActor.HandleNotProcessedFilesFromSDESMessage
-import services.HandleNotProcessedFilesService
+import services.HandleStuckNotificationsService
 
-class HandleNotProcessedFilesFromSDESJob @Inject()(val config: Configuration,
-                                                   val handleNotProcessedFilesService: HandleNotProcessedFilesService,
+class HandleStuckNotificationsJob @Inject()(val config: Configuration,
+                                                   val handleNotProcessedFilesService: HandleStuckNotificationsService,
```
Comment (LewisMitchell) [inline app/scheduler/jobs/HandleNotProcessedFilesFromSDESJob.scala:28]:
```suggestion
                                                   val handleStuckNotificationsService: HandleStuckNotificationsService,
```

File: app/services/HandleStuckNotificationsService.scala [hunk 3]
```diff
@@ -33,14 +33,15 @@ import javax.inject.Inject
...
-                                         appConfig: AppConfig
+class HandleStuckNotificationsService @Inject()(lockRepositoryProvider: MongoLockRepository,
+                                                fileNotificationRepository: FileNotificationRepository,
+                                                timeMachine: TimeMachine,
+                                                config: Configuration,
+                                                appConfig: AppConfig
                                         )(implicit ec: ExecutionContext) extends ScheduledService[Either[ScheduleStatus.JobFailed, String]] {
 
-  val jobName = "HandleNotProcessedFilesFromSDESJob"
+//  val jobName = "HandleStuckNotificationsJob"
```
Comment (LewisMitchell) [inline app/services/HandleStuckNotificationsService.scala:43]: Please remove

File: app/utils/PagerDutyHelper.scala [hunk 4]
```diff
@@ -32,6 +32,7 @@ object PagerDutyHelper {
     final val FAILED_TO_INSERT_FILE_NOTIFICATION = Value
     final val NOTIFICATION_SET_TO_PERMANENT_FAILURE = Value
     final val NOTIFICATION_SET_TO_NOT_PROCESSED_PENDING_RETRY = Value
+    final val NOTIFICATION_SET_TO_NOT_RECEIVED_IN_SDES_PENDING_RETRY = Value
```
Comment (LewisMitchell) [inline app/utils/PagerDutyHelper.scala:35]: Is there an alert config PR for this?

File: app/repositories/FileNotificationRepository.scala [hunk 5]
```diff
@@ -91,7 +91,14 @@ class FileNotificationRepository @Inject()(mongoComponent: MongoComponent,
     collection.find(in("status", Seq(
       RecordStatusEnum.PENDING.toString,
       RecordStatusEnum.NOT_PROCESSED_PENDING_RETRY.toString,
-      RecordStatusEnum.FAILED_PENDING_RETRY.toString): _*
+      RecordStatusEnum.FAILED_PENDING_RETRY.toString,
+      RecordStatusEnum.FILE_NOT_RECEIVED_IN_SDES_PENDING_RETRY.toString): _*
+    )).toFuture()
+  }
+
+  def getNotificationsSentSDES(): Future[Seq[SDESNotificationRecord]] = {
```
Comment (LewisMitchell) [inline app/repositories/FileNotificationRepository.scala:99]: Could this method be made generic so it takes a status as a parameter?
```suggestion
  def getNotificationsByStatus(status: RecordStatusEnum.Value): Future[Seq[SDESNotificationRecord]] = {
```

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/utils/PagerDutyHelper.scala (hunk 4 above)
Comment (AndrewRButcher) [reply app/utils/PagerDutyHelper.scala:35]: Raised https://github.com/hmrc/alert-config/pull/2974

**Review by LewisMitchell — APPROVED** [review]

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**